src/api/libcellml/analyser.h

```diff
@@ -0,0 +1,75 @@
+/*
+Copyright libCellML Contributors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+#pragma once
+
+#include "libcellml/logger.h"
+
+namespace libcellml {
+
+/**
+ * @brief The Analyser class.
+ *
+ * The Analyser class is for representing a CellML Analyser, which aim is to
+ * check whether a model makes mathematical sense. If a model makes mathematical
+ * sense then an @c AnalyserModel object can be retrieved, which can be used to
+ * generate code, for instance.
+ */
+class LIBCELLML_EXPORT Analyser: public Logger
+{
+public:
+    ~Analyser() override; /**< Destructor */
+    Analyser(const Analyser &rhs) = delete; /**< Copy constructor */
+    Analyser(Analyser &&rhs) noexcept = delete; /**< Move constructor */
+    Analyser &operator=(Analyser rhs) = delete; /**< Assignment operator */
+
+    /**
+     * @brief Create an @c Analyser object.
+     *
+     * Factory method to create an @c Analyser.  Create an analyser with::
+     *
+     *   AnalyserPtr analyser = libcellml::Analyser::create();
+     *
+     * @return A smart pointer to an @c Analyser object.
+     */
+    static AnalyserPtr create() noexcept;
+
+    /**
+     * @brief Analyse the @c Model.
+     *
+     * Analyse the @c Model using this @c Analyser.
+     *
+     * @param model The @c Model to analyse.
+     */
+    void analyseModel(const ModelPtr &model);
+
+    /**
+     * @brief Get the analysed model.
+     *
+     * Get the analysed model that is the result of the model analysis.
+     *
+     * @return The analysed model for the @c Model analysed by this @c Analyser.
+     */
+    AnalyserModelPtr model() const;
```

Comment:
Apologies if this has already been discussed, I feel it has, but why is this method not called `analyserModel()`?

Reply:
This hadn't been discussed before, but I can see why you would want it to be called that way. Ok, will do that (and something similar for `AnalyserVariable`, `AnalyserEquation`, etc.).

Reply:
Ping @hsorby, @kerimoyle and @nickerso since I edited my [previous comment](https://github.com/cellml/libcellml/pull/631/files/40a3682e91c766577e1dce017c98c81b0461afc9..cf5cc4eab17d7797fdbc1d3188d598023ec55367#r478643403) a few times and I am not sure whether GitHub sends an email everytime someone edits an existing comment (I don't believe it does).

Reply:
I agree with your point here but I do not like the fact that you are asking for a model.  A model is already a well known thing in libCellML.  I would indeed prefer this to return something like `xxx` obviously can't think of a proper name right now. Now that you bring it up I also don't like the name of the `AnalyserModel` class as I really don't feel that it is a model of an analyser.

Reply:
Yes, I came up with `AnalyserModel` for lack of a better name. Open to suggestions though...

Reply:
`AbstractModel`

Reply:
Probably not the best place for brainstorming a name.

Reply:
I would like to defer this to a separate issue and rename probably all of these `Analyser*` classes.

Reply:
Ok, will then consider this conversation as resolved, for now at least.

Reply:
Can you open an issue for this so we don't forget please?

Reply:
for reference: https://github.com/cellml/libcellml/issues/722